tests/pytest_tests/conftest.py

```diff
@@ -20,6 +21,50 @@
 from wandb.sdk.lib.gitlib import GitRepo  # noqa: E402
 from wandb.sdk.lib.paths import StrPath  # noqa: E402
 
+# --------------------------------
+# Global pytest configuration
+# --------------------------------
+
+
+@pytest.fixture(autouse=True)
+def toggle_wandb_core(
+    monkeypatch: pytest.MonkeyPatch,
+    request: pytest.FixtureRequest,
+) -> None:
+    """Sets WANDB__REQUIRE_CORE in each test.
+
+    wandb-core is a Go rewrite of some Python logic, and all tests should work
+    the same both with and without it.
+    """
+    monkeypatch.setenv("WANDB__REQUIRE_CORE", str(request.param))
+
+
+def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
+    # See https://docs.pytest.org/en/7.1.x/how-to/parametrize.html#basic-pytest-generate-tests-example
+
+    # Run each test both with and without wandb-core.
+    if toggle_wandb_core.__name__ in metafunc.fixturenames:
+        # Allow tests to opt-out of wandb-core until we have feature parity.
+        skip_wandb_core = False
+        for mark in metafunc.definition.iter_markers():
+            if mark.name == "wandb_core_failure":
+                skip_wandb_core = True
+                break
```

Comment:
is there a better place to do it? seems kinda hack-y

Reply:
i think this is how marks are meant to be used---`iter_markers()` is how plugins consume marks

Reply:
ok